src/components/admin/Billing.tsx

```diff
@@ -0,0 +1,460 @@
+                            </Select>
+                        </FormControl>
+                    ) : null}
+
+                    {selectedTenant && methods.length > 0 && !methodsLoading ? (
+                        <TableContainer>
+                            <Table
+                                sx={{ minWidth: 650 }}
+                                aria-label="simple table"
+                                size="small"
+                            >
+                                <TableHead>
+                                    <TableRow>
+                                        <TableCell width={200}>Type</TableCell>
+                                        <TableCell>Name</TableCell>
+                                        <TableCell>Last 4 digits</TableCell>
+                                        <TableCell>Details</TableCell>
+                                        <TableCell>Primary</TableCell>
+                                        <TableCell>Actions</TableCell>
+                                    </TableRow>
+                                </TableHead>
+                                <TableBody>
+                                    {methods.map((method) => (
+                                        <PaymentMethod
+                                            onDelete={async () => {
+                                                await deleteTenantPaymentMethod(
+                                                    selectedTenant,
+                                                    method.id
+                                                );
+                                                setRefreshCounter((r) => r + 1);
+                                            }}
+                                            onPrimary={async () => {
+                                                await setTenantPrimaryPaymentMethod(
+                                                    selectedTenant,
+                                                    method.id
+                                                );
+                                                setRefreshCounter((r) => r + 1);
+                                            }}
+                                            key={method.id}
+                                            {...method}
+                                            primary={
+                                                method.id === defaultSource
+                                            }
+                                        />
+                                    ))}
+                                </TableBody>
+                            </Table>
+                        </TableContainer>
+                    ) : methodsLoading ? (
+                        <>
+                            <Skeleton animation="wave" />
+                            <Skeleton animation="wave" />
+                            <Skeleton animation="wave" />
+                        </>
```

Comment:
Please replace this with the standard table loading component `TableLoadingRows`

Reply:
We talked in a DM about this: I should be using the Entity Table component, but you're working on making it more flexible and will update this table as part of that work